airflow/contrib/hooks/livy_hook.py

```diff
@@ -0,0 +1,379 @@
+        """
+        Wrapper for HttpHook, allows to change method on the same HttpHook
+
+        :param method: http method
+        :type method: str
+        :param endpoint: endpoint
+        :type endpoint: str
+        :param data: request payload
+        :type data: dict
+        :param headers: headers
+        :type headers: dict
+        :param extra_options: extra options
+        :type extra_options: dict
+        :return: http response
+        :rtype: requests.Response
+        """
+        if method not in ('GET', 'POST', 'PUT', 'DELETE', 'HEAD'):
+            raise AirflowException("Invalid http method '{}'".format(method))
+
+        back_method = self.method
+        self.method = method
+        try:
+            result = self.run(endpoint, data, headers, extra_options)
+        finally:
+            self.method = back_method
+        return result
+
+    def post_batch(self, *args, **kwargs):
+        """
+        Perform request to submit batch
+
+        :return: batch session id
+        :rtype: int
+        """
+        batch_submit_body = json.dumps(self.build_post_batch_body(*args, **kwargs))
+
+        if self.base_url is None:
+            # need to init self.base_url
+            self.get_conn()
+        self.log.info("Submitting job {} to {}".format(batch_submit_body, self.base_url))
+
+        response = self.run_method(
+            method='POST',
+            endpoint='/batches',
+            data=batch_submit_body
+        )
+        self.log.debug("Got response: {}".format(response.text))
+
+        if response.status_code != 201:
+            raise AirflowException("Could not submit batch. Status code: {}".format(response.status_code))
+
+        batch_id = self._parse_post_response(response.json())
+        if batch_id is None:
+            raise AirflowException("Unable to parse a batch session id")
+        self.log.info("Batch submitted with session id: {}".format(batch_id))
+
+        return batch_id
+
+    def get_batch(self, session_id):
+        """
+        Fetch info about the specified batch
+
+        :param session_id: identifier of the batch sessions
+        :type session_id: int
+        :return: response body
+        :rtype: dict
+        """
+        self._validate_session_id(session_id)
+
+        self.log.debug("Fetching info for batch session {}".format(session_id))
+        response = self.run_method(endpoint='/batches/{}'.format(session_id))
+
+        if response.status_code != 200:
+            self.log.warning("Got status code {} for session {}".format(response.status_code, session_id))
+            raise AirflowException("Unable to fetch batch with id: {}".format(session_id))
+
+        return response.json()
+
+    def get_batch_state(self, session_id):
+        """
+        Fetch the state of the specified batch
+
+        :param session_id: identifier of the batch sessions
+        :type session_id: Union[int, str]
```

Comment:
```suggestion
        :type session_id: int or str
```

Doc comments don't use the same type syntax as code, much to our annoyance.